src/parsing.rs

```diff
@@ -106,14 +111,31 @@ fn read_n_bytes<R: Read>(file: &mut R, length: usize) -> io::Result<Vec<u8>> {
 
 #[allow(clippy::vec_box)]
 fn read_constant_pool<R: Read>(file: &mut R) -> io::Result<Vec<Box<ConstantPoolEntry>>> {
-    let constant_pool_count = i32::from(read_u16(file)?);
+    let mut constant_pool_count = read_u16(file)? - 1;
 
-    let mut constant_pool = Vec::<Box<ConstantPoolEntry>>::new();
+    let mut constant_pool = Vec::<Box<ConstantPoolEntry>>::with_capacity(constant_pool_count as usize);
 
-    for _ in 0..(constant_pool_count - 1) {
+    while constant_pool_count > 0 {
```

Comment:
Instead of mutating `constant_pool_count`, can you make a new mutable variable to use for this loop (ex. `i` or `current_constant_index`)? It looks like the mutation is only needed for the context of the loop, so using a new variable for it might make that clearer.

Reply:
fixed